lib/Viroverse/ISLAWorksheet.pm

```diff
@@ -0,0 +1,260 @@
+use strict;
+use warnings;
+use utf8;
+
+package Viroverse::ISLAWorksheet;
+use Moo;
+use Types::Standard qw< :types >;
+use Viroverse::Logger qw< :log >;
+
+=head1 NAME
+
+Viroverse::ISLAWorksheet - convert between a Viroverse::Result::Sample and an
+Excel spreadsheet for ISLA data
+
+=head1 ATTRIBUTES
+
+=head2 model
+
+The Viroverse::Model::sample instance that this spreadsheet represents
+
+=head1 METHODS
+
+=head2 make_xlsx
+
+Generates an Excel spreadsheet (in xlsx format) that can be filled out with
+added data for the sample. Returns a FileHandle that has been seeked to the
+beginning.
+
+=cut
+
+has model => (
+    is       => 'ro',
+    isa      => Object,
+    required => 1,
+);
+
+sub make_xlsx {
```

Comment:
In the existing usage there's not necessarily a real impact, but you can get memoization of the resulting xlsx by making `xlsx` a `lazy` attribute (`has xlsx => ( is => 'lazy', isa => Something);`) and renaming this method to `_build_xlsx`.

Reply:
I don't really like the thought of this because it makes some implications regarding idempotency; if someone were to upload a new spreadsheet that gets processed, which spreadsheet gets returned by this attribute?

Reply:
This is just a spreadsheet-generating component for now, which already always returns equivalent output data given the same sample as input. I don't think there's an issue with idempotency here. If we adapt this object to consume as well as produce spreadsheets we'll have to rework the interface anyway.

Reply:
okay I changed it to a lazy property